### PR TITLE
New module: digital_ocean_snapshot_facts

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pmarques Could you please review this ?

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            has_next = "pages" in response.json["links"] and "next" in response.json["links"]["pages"]
+
+    if status_code != 200:
+        module.fail_json(msg="Failed to fetch snapshot facts due to error : %s" % response.json['message'])
+
+    module.exit_json(changed=False, data=snapshot)
+
+
+def main():
+    module = AnsibleModule(
```
Comment (pmarques) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py]: A few days ago **module_utils/digital_ocean.py** (Ansible Digital Ocean utils) was updated in PR #35934 which introduced generic configurations and documentation that can be reused, namely this function **digital_ocean_argument_spec**(https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/digital_ocean.py)

Comment (pmarques) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py]: A few days ago **lib/ansible/module_utils/digital_ocean.py** (Ansible Digital Ocean utils) was updated in PR #35934 which introduced generic configurations and documentation that should be used, particularly here this function **digital_ocean_argument_spec**(https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/digital_ocean.py)

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    default: 'all'
+    choices: [ 'all', 'droplet', 'volume', 'by_id']
+    required: false
+  snapshot_id:
+    description:
+     - To retrieve information about a snapshot, please specify this as a snapshot id.
+     - If set to actual snapshot id, then facts are gathered related to that particular snapshot only.
+     - This is required parameter, if C(snapshot_type) is set to C(by_id).
+    required: false
+  oauth_token:
```
Comment (pmarques) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py]: A few days ago **utils/module_docs_fragments/digital_ocean.py** (Ansible Digital Ocean utils) was updated in PR #35934 which introduced generic configurations and documentation that should be used, particularly here this documentation fragment **extends_documentation_fragment: digital_ocean.documentation**(https://github.com/Akasurde/ansible/blob/devel/lib/ansible/utils/module_docs_fragments/digital_ocean.py)

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+def core(module):
+    snapshot_type = module.params['snapshot_type']
+
+    rest = DigitalOceanHelper(module)
+
+    # Check if api_token is valid or not
+    response = rest.get('account')
+    if response.status_code == 401:
+        module.fail_json(msg='Failed to login using api_token, please verify '
+                             'validity of api_token')
```
Comment (pmarques) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py]: AFAIK we can remove this (last 4 lines) since is part of the DO helper?

**Review by pmarques — COMMENTED** [review]
Some comments inline, apart from that seems to be ok.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py (hunk 3 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py]: Done.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py]: Done.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py]: Done.

**Comment by BondAnthony** [comment]
LGTM

Thank you!

**Comment by Akasurde** [comment]
@BondAnthony @pmarques Thanks for your views.

**Pull request merged**